### ISSUE TITLE
Installing dotnet sdk 6.0.414 should install dotnet runtime 6.0.22 and not 6.0.23

### ISSUE THREAD

**Issue by sMuro** [dotnet/core#8841]
### Description

Tested both 20.04 and 22.04 repositories and the behavior is same.

On Ubuntu 20.04: 
Install dotnet-sdk-6.0 version 6.0.414-1
Actual: dotnet-runtime-6.0 version 6.0.23-1 is installed as dependency.
Expected: dotnet-runtime-6.0 version 6.0.22-1 to be installed as dependency.

```bash
# apt install dotnet-sdk-6.0=6.0.414-1
# dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.414
 Commit:    dc3b5c4341

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /usr/share/dotnet/sdk/6.0.414/

global.json file:
  Not found

Host:
  Version:      6.0.23
  Architecture: x64
  Commit:       e0f0de876a

.NET SDKs installed:
  6.0.414 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.23 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.23 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info
```

### Configuration

Ubuntu 20.04.6 LTS 
x64

**Comment by NikolaMilosavljevic** [comment]
This seems expected. `dotnet-sdk-6.0.404` package does not specify specific version of dependencies. Ubuntu package manager would install the highest available, with minimum `6.0.12`. Here's the relevant output of `dpkg-deb -I` command:

 _**Depends: dotnet-targeting-pack-6.0 (>= 6.0.12), netstandard-targeting-pack-2.1 (>= 2.1.0), aspnetcore-runtime-6.0 (>= 6.0.12), dotnet-apphost-pack-6.0 (>= 6.0.12), dotnet-runtime-6.0 (>= 6.0.12), aspnetcore-targeting-pack-6.0 (>= 6.0.12)**_

cc @joeloff

**Comment by sMuro** [comment]
Hello, I also tried to install dotnet runtime 6.0.22 specifically, but then it removed the dotnet-sdk-6.0 installation alltogether. Thus I cannot install 6.0.414 sdk version with dotnet runtime 6.0.22. This might cause issues in my usecase. Thanks.
